script.js

```diff
@@ -0,0 +1,245 @@
+const allButtons = document.getElementsByTagName("button");
+const numberButtons = document.querySelectorAll("[data-number]");
+const operationButtons = document.querySelectorAll("[data-operation]");
+const equalsButton = document.querySelector("[data-equals]");
+const clearButton = document.querySelector("[data-clear]");
+const deleteButton = document.querySelector("[data-delete]");
+const plusminus = document.querySelector("[data-plusminus]");
+const inputNumberField = document.querySelector(".inputNumber");
+
+let inputOperator = undefined;
+let savedInputNumber = undefined;
+let currentNumber = 0;
+let resultNumber = undefined;
+
+let replaceNumber = true;
+
+document.body.onkeypress = (e) => {
```

Comment:
`onkeypress` is deprecated. What was the reason to not put this logic in the `onkeydown`?

Reply:
onkeydown detects inputs like "F1 - F12" but onkeypress doesn't. Before, I didn't know how to tackle this problem correctly so I used it as a workaround.

However, I was able to remove the use of onkeypress now by updating the "isDigit" function.